File: lec_select.py
import sqlite3 as sql                         # sqlite module

con = sql.connect( 'firefly.sqlite' )        # open database
cur = con.cursor()                        

cur.execute( '''SELECT * FROM staff;''' )    # run query
for row in cur:                                # loop over results
    print( row )
    
con.close()                                    # close database

File: lec_select.cpp
#include <iostream>
using namespace std;

#include "libsqlite.hpp"                    // sqlite library

int main()
{
    sqlite::sqlite db( "firefly.sqlite" );    // open database

    auto cur = db.get_statement();            // create query
    cur->set_sql( "SELECT * FROM staff;" );
    cur->prepare();                            // run query

    while( cur->step() )                    // loop over results
        cout << cur->get_int(0) << " " << cur->get_text(1) << endl;
}